icd.c

```diff
@@ -198,6 +198,12 @@ void khrIcdVendorAdd(const char *libraryName)
 
 void khrIcdContextPropertiesGetPlatform(const cl_context_properties *properties, cl_platform_id *outPlatform)
 {
+    if (properties == NULL && khrIcdVendors != NULL)
```

Comment:
from spec of clCreateContext, CL_INVALID_PLATFROM is returned for two cases:
- if properties is NULL and on platform could be selected
- if platform value specified in properties is not a valid platform.

it does not touch the case that properties is non-NULL but does not contain a CL_CONTEXT_PLATFORM property. My understanding is to select an implementation-defined platform for such case.

As in this code review, as the subject suggested, it is for the case that properties is NULL, maybe we can firstly merge this change, and then add another change for the case that properties is not NULL.